lib/rubocop/cop/naming/rescued_exceptions_variable_name.rb

```diff
@@ -76,6 +76,19 @@ def location
         def message(_node = nil)
           format(MSG, preferred: preferred_name, bad: variable_name)
         end
+
+        def correct_code?
+          @exception_name.const_type? ||
+            @exception_name.splat_type? ||
+            variable_name == preferred_name ||
+            exception_with_method?
+        end
+
+        def exception_with_method?
+          return unless @exception_name.children.first.respond_to?(:const_type?)
+
+          @exception_name.children.first.const_type?
+        end
```

Comment:
I'm not proud if this method... It feels hacky.

Am I right writing this ? Is there a way to clean it ?

Reply:
Generally for situations like this it's much better to use just node pattern matching. Take a look at the docs or the code for other cops. With the right pattern you can potentially eliminate this method and `correct_code?` altogether.